packages/autoloader/src/AutoloadProcessor.php

```diff
@@ -88,6 +88,22 @@ public function processClassmap( $autoloads, $scanPsrPackages ) {
 			}
 		}
 
+		if ( ! empty( $autoloads['psr-0'] ) ) {
```

Comment:
Why does this always scan, when psr-4 packages are only scanned if the caller set `$scanPsrPackages` true?

Reply:
Prior to #16819 the autoloader would always convert namespaces into classmaps regardless of optimization level. While we now support unoptimized PSR-4 its worth noting that we don't support PSR-0 and so anything other than this classmap export at the moment would do nothing.

Reply:
Let me see if I understand this correctly (I'm new):

* Originally it always read PSR-4 classes into a classmap.
* That caused problems for developers, because new PSR-4 classes weren't being picked up as expected without a composer run to regenerate everything. So #16819 was created to make the "read into classmap" only happen with --optimize-autoloader.
* Now we're adding PSR-0 support, with the same problem for developers that was fixed for PSR-4 in #16819.

I'm guessing we don't care about fixing that developer issue for PSR-0 because we're only adding PSR-0 support for use with external libraries, and our developers should always be using PSR-4? That seems valid to me, but perhaps could use a comment in the code explaining the different handling.

Reply:
> That seems valid to me, but perhaps could use a comment in the code explaining the different handling.

Good point! I'll add a comment.